### PR TITLE
ARROW-8245: [Python][Parquet] Skip hidden directories when reading partitioned parquet files

### PR DESCRIPTION
**Pull request by caleboverman**

### CODE REVIEW

File: python/pyarrow/tests/test_parquet.py [hunk 1]
```diff
@@ -2337,7 +2337,22 @@ def _make_example_multifile_dataset(base_path, nfiles=10, file_nrows=5):
...
-def test_ignore_private_directories(tempdir):
+def test_ignore_private_directories_dot(tempdir):
+    dirpath = tempdir / guid()
+    dirpath.mkdir()
+
+    paths = _make_example_multifile_dataset(dirpath, nfiles=10,
+                                            file_nrows=5)
+
+    # private directory
+    (dirpath / '.spark-staging').mkdir()
```
Comment (wesm) [inline python/pyarrow/tests/test_parquet.py:2348]: Perhaps we can parametrize a single private directory test with a list of different private directories and avoid some code duplication?

Comment (jorisvandenbossche) [reply python/pyarrow/tests/test_parquet.py:2348]: Yes, this test and the one below can indeed easily be combined

**Review by jorisvandenbossche — COMMENTED** [review]
Pending the test comment, looks good to me!

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by wesm** [comment]
Merging. The test failure is ARROW-8323

**Comment by jorisvandenbossche** [comment]
@caleboverman thanks for the quick PR!

**Comment by caleboverman** [comment]
@jorisvandenbossche thank you for all you do with arrow. i'm interested in opportunities in helping out with parquet/python issues going forward.

**Comment by jorisvandenbossche** [comment]
@caleboverman there are several open python and parquet issues on JIRA, so very welcome to look into that. Specifically for Parquet, what will also be very valuable is actually testing the new datasets API. In the upcoming pyarrow 0.17 release, the `parquet.read_table`/`ParquetDataset` will have a `use_legacy_dataset=False` option to use the revamped implementation. This is not yet the default, but battles testing this on your applications will be very useful.

**Pull request closed without merging**